src/httprpc.cpp

```diff
@@ -180,21 +180,22 @@ static bool HTTPReq_JSONRPC(HTTPRequest* req, const std::string &)
         // Set the URI
         jreq.URI = req->GetURI();
 
-        std::string strReply;
-        // singleton request
+        UniValue ret;
         if (valRequest.isObject()) {
-            jreq.parse(valRequest);
-
-            UniValue result = tableRPC.execute(jreq);
+            ret = JSONRPCExecOne(jreq, valRequest);
+        } else if (valRequest.isArray()) {
+            ret = JSONRPCExecBatch(jreq, valRequest.get_array());
+        } else {
+            throw JSONRPCError(RPC_PARSE_ERROR, "Top-level object parse error");
+        }
 
-            // Send reply
-            strReply = JSONRPCReply(result, NullUniValue, jreq.id);
+        if (ret.isNull()) {
+            // JSON-RPC 2.0: return nothing at all
+            return false;
```

Comment:
Does this create different behavior for degenerate returns when not passing `-strictjsonrpcspec`?